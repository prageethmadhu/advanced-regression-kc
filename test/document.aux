\relax 
\providecommand\hyper@newdestlabel[2]{}
\providecommand\HyField@AuxAddToFields[1]{}
\providecommand\HyField@AuxAddToCoFields[2]{}
\@writefile{nav}{\headcommand {\slideentry {0}{0}{1}{1/1}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {1}{1}}}
\@writefile{nav}{\headcommand {\slideentry {0}{0}{2}{2/2}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {2}{2}}}
\@writefile{nav}{\headcommand {\slideentry {0}{0}{3}{3/3}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {3}{3}}}
\@writefile{toc}{\beamer@sectionintoc {1}{Starting Simple}{4}{0}{1}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {1}{3}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {1}{3}}}
\@writefile{nav}{\headcommand {\sectionentry {1}{Starting Simple}{4}{Starting Simple}{0}}}
\@writefile{nav}{\headcommand {\slideentry {1}{0}{1}{4/4}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {4}{4}}}
\@writefile{nav}{\headcommand {\slideentry {1}{0}{2}{5/5}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {5}{5}}}
\newlabel{fig:price_vs_size_fit}{{1}{6}{Price vs. Size Fit}{fig:price_vs_size_fit}{}}
\@writefile{snm}{\beamer@slide {fig:price_vs_size_fit}{6}}
\@writefile{nav}{\headcommand {\slideentry {1}{0}{3}{6/6}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {6}{6}}}
\@writefile{nav}{\headcommand {\slideentry {1}{0}{4}{7/7}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {7}{7}}}
\@writefile{nav}{\headcommand {\slideentry {1}{0}{5}{8/8}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {8}{8}}}
\@writefile{toc}{\beamer@sectionintoc {2}{Expanding the Scope}{9}{0}{2}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {4}{8}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {4}{8}}}
\@writefile{nav}{\headcommand {\sectionentry {2}{Expanding the Scope}{9}{Expanding the Scope}{0}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{1}{9/9}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {9}{9}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{2}{10/10}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {10}{10}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{3}{11/11}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {11}{11}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{4}{12/12}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {12}{12}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{5}{13/13}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {13}{13}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{6}{14/14}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {14}{14}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{7}{15/15}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {15}{15}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{8}{16/16}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {16}{16}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{9}{17/17}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {17}{17}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{10}{18/18}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {18}{18}}}
\@writefile{lol}{\contentsline {lstlisting}{##}{19}{}\protected@file@percent }
\@writefile{nav}{\headcommand {\slideentry {2}{0}{11}{19/19}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {19}{19}}}
\@writefile{nav}{\headcommand {\slideentry {2}{0}{12}{20/20}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {20}{20}}}
\@writefile{toc}{\beamer@sectionintoc {3}{Refining Precision}{21}{0}{3}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {9}{20}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {9}{20}}}
\@writefile{nav}{\headcommand {\sectionentry {3}{Refining Precision}{21}{Refining Precision}{0}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{1}{21/21}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {21}{21}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{2}{22/22}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {22}{22}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{3}{23/23}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {23}{23}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{4}{24/24}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {24}{24}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{5}{25/25}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {25}{25}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{6}{26/26}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {26}{26}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{7}{27/27}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {27}{27}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{8}{28/28}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {28}{28}}}
\@writefile{nav}{\headcommand {\slideentry {3}{0}{9}{29/29}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {29}{29}}}
\@writefile{toc}{\beamer@sectionintoc {4}{Facing Challenges}{30}{0}{4}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {21}{29}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {21}{29}}}
\@writefile{nav}{\headcommand {\sectionentry {4}{Facing Challenges}{30}{Facing Challenges}{0}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{1}{30/30}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {30}{30}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{2}{31/31}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {31}{31}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{3}{32/32}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {32}{32}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{4}{33/33}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {33}{33}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{5}{34/34}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {34}{34}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{6}{35/35}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {35}{35}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{7}{36/36}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {36}{36}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{8}{37/37}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {37}{37}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{9}{38/38}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {38}{38}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{10}{39/39}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {39}{39}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{11}{40/40}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {40}{40}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{12}{41/41}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {41}{41}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{13}{42/42}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {42}{42}}}
\@writefile{nav}{\headcommand {\slideentry {4}{0}{14}{43/43}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {43}{43}}}
\@writefile{toc}{\beamer@sectionintoc {5}{Insights \& Next Steps}{44}{0}{5}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {30}{43}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {30}{43}}}
\@writefile{nav}{\headcommand {\sectionentry {5}{Insights \& Next Steps}{44}{Insights \& Next Steps}{0}}}
\@writefile{nav}{\headcommand {\slideentry {5}{0}{1}{44/44}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {44}{44}}}
\@writefile{nav}{\headcommand {\slideentry {5}{0}{2}{45/45}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {45}{45}}}
\@writefile{nav}{\headcommand {\slideentry {5}{0}{3}{46/46}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {46}{46}}}
\@writefile{nav}{\headcommand {\slideentry {5}{0}{4}{47/47}{}{0}}}
\@writefile{nav}{\headcommand {\beamer@framepages {47}{47}}}
\@writefile{nav}{\headcommand {\beamer@partpages {1}{47}}}
\@writefile{nav}{\headcommand {\beamer@subsectionpages {44}{47}}}
\@writefile{nav}{\headcommand {\beamer@sectionpages {44}{47}}}
\@writefile{nav}{\headcommand {\beamer@documentpages {47}}}
\@writefile{nav}{\headcommand {\gdef \inserttotalframenumber {42}}}
\gdef \@abspage@last{47}
