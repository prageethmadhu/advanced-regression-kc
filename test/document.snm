\beamer@slide {fig:price_vs_size_fit}{5}
