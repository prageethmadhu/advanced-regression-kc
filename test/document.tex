\documentclass{beamer}
\usetheme{metropolis}
\usecolortheme{spruce}
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{caption}
\usepackage{listings}
\lstset{
	basicstyle=\ttfamily\tiny, % Adjusted for richer content
	breaklines=true,
	frame=single,
	numbers=left,
	numberstyle=\tiny\color{gray},
	backgroundcolor=\color{gray!10},
	keywordstyle=\color{blue},
	stringstyle=\color{red},
	commentstyle=\color{green!60!black},
	language=R % Default, switches to Python
}

% Title slide information
\title{Regression \& Prediction}
\subtitle{Theory and Practice with House Prices}
\author{Your Name}
\institute{xAI}
\date{February 23, 2025}

\begin{document}
	
	% Title Slide
	\begin{frame}
		\titlepage
	\end{frame}
	
	% Outline Slide
	\begin{frame}{Our Housing Journey}
		\begin{columns}
			\column{0.5\textwidth}
			\begin{itemize}
				\item Starting Simple: Foundations
				\item Expanding the Scope: Complexity
				\item Refining Precision: Optimization
			\end{itemize}
			\column{0.5\textwidth}
			\begin{itemize}
				\item Facing Challenges: Pitfalls
				\item Insights \& Next Steps: Applications
			\end{itemize}
		\end{columns}
		\metroset{block=fill}
		\begin{block}{Focus}
			Unpacking King County house prices with rich theory and dual R/Python implementations
		\end{block}
	\end{frame}
	
	% Section 1: Starting Simple
	\section{Starting Simple}
	
	\begin{frame}{The Housing Puzzle}
		\begin{itemize}
			\item \textbf{Objective}: Decode drivers of house prices in King County—size, location, features
			\item \textbf{Regression}: A statistical lens linking price ($Y$) to predictors like size ($X$)
			\item \textbf{Purpose}: Explain historical sales patterns and forecast future values for buyers and assessors
		\end{itemize}
	\end{frame}
	
	\begin{frame}[fragile]{Simple Linear Regression: Theory \& R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 141)}: Models a straight-line relationship: $Y = b_0 + b_1X + e$
				\item $b_0$: Base price when size is zero; $b_1$: Price increase per sq ft; $e$: Random error
				\item Assumes linearity and independence—foundation of regression
			\end{itemize}
			\begin{lstlisting}
# R (p. 152 adapted)
simple_lm <- lm(AdjSalePrice ~ SqFtTotLiving, data = house)
# Output: b_0 ~ base, b_1 ~ price/sq ft
			\end{lstlisting}
			\column{0.4\textwidth}
		\begin{figure}
			\centering
			\includegraphics[width=\textwidth]{regres1.jpg} % Use .png, .jpg, or .jpeg instead of .pdf
			\caption{Price vs. Size Fit}
			\label{fig:price_vs_size_fit}
		\end{figure}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Simple Linear Regression: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Fits price to living space, revealing size’s impact
				\item \textbf{Key Insight}: Positive slope shows larger homes fetch higher prices
			\end{itemize}
			\begin{lstlisting}
# Python (p. 152 adapted)
from sklearn.linear_model import LinearRegression
predictors = ['SqFtTotLiving']
outcome = 'AdjSalePrice'
simple_lm = LinearRegression()
simple_lm.fit(house[predictors], house[outcome])
# Example: Intercept ~ base, Coef ~ $ per sq ft
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Size as a core driver
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}{Finding the Best Fit: Least Squares}
		\begin{itemize}
			 \item \textbf{How do we draw that line?} We minimize the mess—sum of squared errors
			\item \textbf{Theory}: Finds the line minimizing residual sum of squares: $\sum_{i=1}^n (Y_i - \hat{Y}_i)^2$
			\item \textbf{How}: Adjusts $b_0$ and $b_1$ to reduce prediction errors—optimal for linear fits
			\item \textbf{History}: Legendre (1805) and Gauss; computationally efficient but outlier-sensitive in small datasets
		\end{itemize}
	\end{frame}
	
	% Section 2: Expanding the Scope
	\section{Expanding the Scope}
	
	\begin{frame}[fragile]{More Clues: Multiple Linear Regression}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory}: Extends to multiple predictors: $Y = b_0 + b_1X_1 + b_2X_2 + \dots + e$
				\item \textbf{Power}: Captures combined effects—size, lot, bedrooms—assuming linearity
				\item \textbf{Use}: Explains complex housing dynamics
			\end{itemize}
			\begin{lstlisting}
# R (p. 152)
house_lm <- lm(AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms +
Bedrooms + BldgGrade, data = house)
# Coefs: SqFtTotLiving = 228.831, Bedrooms = -47769.955
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Size adds \$229/sq ft
			\end{itemize}
		\end{columns}
	\end{frame}
	

% Section 2: Key Findings

\begin{frame}[fragile]{Multiple Linear Regression: Key Findings}
	\begin{columns}
		\column{0.9\textwidth}
		\begin{lstlisting}
lm(formula = price ~ sqft_living + sqft_lot + bathrooms + bathrooms + 
grade, data = house_df, na.action = na.omit)

Coefficients:
(Intercept)  sqft_living     sqft_lot    bathrooms        grade  
-5.957e+05    2.065e+02   -2.664e-01   -3.944e+04    1.037e+05  
		\end{lstlisting}
		\begin{itemize}
			\item \textbf{sqft\_living}: 🏡 +\$206.5 per sq ft → Larger houses increase price significantly.
			\item \textbf{grade}: 🔼 +\$103,700 per unit → Higher quality homes boost price.
			\item \textbf{sqft\_lot}: 🌲 -\$0.266 per sq ft → Lot size has a tiny negative effect.
			\item \textbf{bathrooms}: 🚿 -\$39,440 per extra bathroom → Unexpected negative impact (possible interaction effect).
		\end{itemize}
	\end{columns}
\end{frame}


% Section 3: Model Performance
\begin{frame}[fragile]{Multiple Linear Regression:Model Evaluation}
	\begin{columns}
		\column{0.9\textwidth}
		\begin{lstlisting}
> cat("RMSE:", RMSE, "\n")
RMSE: 249532.2 
> cat("R²:", R_squared, "\n")
R²: 0.5380018 
> cat("P-values:\n")
P-values:
		\end{lstlisting}
		\begin{itemize}
			\item \textbf{RMSE}: 📉 $261,300 → Predictions are off by ~$261K on average.
			\item \textbf{$R^2$}: 📊 0.5406 (54.06%) → The model explains 54% of price variation..
			\item \textbf{P-values}: 🧪 SqFtLot is **not statistically significant (p = 0.323)`
		\end{itemize}
	\end{columns}
\end{frame}

	
	\begin{frame}[fragile]{Multiple Linear Regression: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Models housing with multiple factors
				\item \textbf{Key Insight}: Negative bedroom coef suggests smaller rooms hurt value
			\end{itemize}
			\begin{lstlisting}
# Define predictors and outcome
predictors = ['sqft_living', 'sqft_lot', 'bathrooms', 'grade']
outcome = 'price'

# Fit Multiple Linear Regression Model
house_lm = LinearRegression()
house_lm.fit(house_df[predictors], house_df[outcome])

			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Bedrooms vs. size tension
			\end{itemize}
		\end{columns}
	\end{frame}
	
	% Section 2: Expanding the Scope
	\begin{frame}[fragile]{Factor Variables: Theory \& R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 163)}: Encodes categorical variables (e.g., property type) as binary dummies
				\item \textbf{Why}: Allows non-numeric factors in regression; compares to a reference level
				\item \textbf{Example}: Single-family vs. Townhouse effects
			\end{itemize}
			\begin{lstlisting}
# R (p. 164)
prop_type_dummies <- model.matrix(~ PropertyType - 1, data = house)
# Output: 1 for each type present
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Type shifts price
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Factor Variables: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Integrates property type into price model
				\item \textbf{Key Insight}: Townhouses may differ from single-family homes
			\end{itemize}
			\begin{lstlisting}
# Python (p. 166 adapted)
import pandas as pd
X = pd.get_dummies(house['PropertyType'], drop_first=True)
# Drops first level (e.g., Multiplex) as reference
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Baseline comparison
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Nonlinear Fit: Theory \& Splines in R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 187)}: Nonlinear via polynomial segments joined at knots
				\item \textbf{Why}: Captures diminishing returns—e.g., small homes gain more per sq ft
				\item \textbf{Advantage}: Flexible fit without overfitting like high-order polynomials
			\end{itemize}
			\begin{lstlisting}
# R (p. 190)
library(splines)
knots <- quantile(house_98105$SqFtTotLiving, p = c(.25, .5, .75))
lm_spline <- lm(AdjSalePrice ~ bs(SqFtTotLiving, knots = knots, degree = 3) +
SqFtLot + Bathrooms + Bedrooms + BldgGrade, data = house_98105)
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{figure}
				\includegraphics[width=\textwidth]{figure4-12-placeholder.pdf}
				\caption{Spline Fit}
			\end{figure}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Nonlinear Fit: Splines in Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Fits nonlinear price trends in zip 98105
				\item \textbf{Key Insight}: Better matches small vs. large home value shifts
			\end{itemize}
			\begin{lstlisting}
# Python (p. 190)
import statsmodels.formula.api as smf
formula = 'AdjSalePrice ~ bs(SqFtTotLiving, df=6, degree=3) + SqFtLot + Bathrooms + Bedrooms + BldgGrade'
model_spline = smf.ols(formula=formula, data=house_98105).fit()
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Curves reflect reality
			\end{itemize}
		\end{columns}
	\end{frame}

	
	

	
	% Section 3: Refining Precision
	\section{Refining Precision}
	
	

	
	\begin{frame}{Model Assessment: Theory}
		\begin{itemize}
			\item \textbf{Theory (p. 153)}: Measures prediction quality and fit
			\item \textbf{RMSE}: $\sqrt{\frac{\sum (y_i - \hat{y}_i)^2}{n}}$—average error magnitude
			\item \textbf{$R^2$}: Proportion of variance explained (0-1); higher means better fit
			\item \textbf{Use}: Guides housing prediction accuracy
		\end{itemize}
	\end{frame}
	
	\begin{frame}{Cross-Validation: Theory}
		\begin{itemize}
			\item \textbf{Theory (p. 155)}: Validates model on unseen data via $k$-fold splits
			\item \textbf{Process}: Divide data, train on $k-1$, test on 1, repeat, average RMSE
			\item \textbf{Why}: Ensures predictions generalize beyond training sales—crucial for real estate
		\end{itemize}
	\end{frame}
	
	\begin{frame}[fragile]{Model Selection: Theory \& R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 156)}: Balances fit vs. complexity—Occam’s razor
				\item \textbf{AIC}: $2P + n\log(\text{RSS}/n)$—penalizes extra predictors
				\item \textbf{Goal}: Optimal housing model without overkill
			\end{itemize}
			\begin{lstlisting}
# R (p. 157)
library(MASS)
house_full <- lm(AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms +
Bedrooms + BldgGrade + PropertyType, data = house)
step <- stepAIC(house_full, direction = "both")
# Drops less impactful vars
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Streamlined predictors
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Model Selection: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Automates predictor choice for housing
				\item \textbf{Key Insight}: Reduces noise, enhances prediction
			\end{itemize}
			\begin{lstlisting}
# Python (p. 158 adapted)
from dmba import stepwise_selection
predictors = ['SqFtTotLiving', 'SqFtLot', 'Bathrooms', 'Bedrooms', 'BldgGrade']
def train_model(vars):
model = LinearRegression()
model.fit(house[vars], house[outcome])
return model
best_model, _ = stepwise_selection(house[predictors].columns, train_model)
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Focused fit
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Weighted Regression: Theory \& R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 159)}: Weights adjust influence by reliability
				\item \textbf{Why}: Older sales less relevant—recent data gets priority
				\item \textbf{Impact}: Refines coefficients for current market
			\end{itemize}
			\begin{lstlisting}
# R (p. 159)
house$Weight = year(house$DocumentDate) - 2005
house_wt <- lm(AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms +
Bedrooms + BldgGrade, data = house, weight = Weight)
# Shifts coefs slightly
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Recent sales emphasized
			\end{itemize}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Weighted Regression: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Weights tune housing model
				\item \textbf{Key Insight}: Aligns predictions with market trends
			\end{itemize}
			\begin{lstlisting}
# Python (p. 160)
house['Weight'] = [int(date.split('-')[0]) for date in house.DocumentDate] - 2005
house_wt = LinearRegression()
house_wt.fit(house[predictors], house[outcome], sample_weight=house.Weight)
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Fresher focus
			\end{itemize}
		\end{columns}
	\end{frame}
	
	% Section 4: Facing Challenges
	\section{Facing Challenges}
	
	\begin{frame}{Prediction Limits: Theory}
		\begin{itemize}
			\item \textbf{Theory (p. 161)}: Extrapolation beyond data fails—e.g., empty lots
			\item \textbf{Intervals}: Confidence for $b_i$, wider prediction for $\hat{Y}_i$
			\item \textbf{Why}: Uncertainty spikes outside training range—limits housing forecasts
		\end{itemize}
	\end{frame}
	
	\begin{frame}{Interpreting Coefficients: Theory}
		\begin{itemize}
			\item \textbf{Theory (p. 171)}: Coefficients mislead if predictors correlate
			\item \textbf{Multicollinearity}: Size and bedrooms overlap—unstable fits
			\item \textbf{Confounding}: Missing location skews results
			\item \textbf{Interactions}: Size’s effect varies by zip—needs modeling
		\end{itemize}
	\end{frame}
	
	\begin{frame}[fragile]{Diagnostics: Theory \& R}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Theory (p. 176)}: Residuals reveal model flaws
				\item \textbf{Outliers}: Extreme sales (e.g., \$119,748); \textbf{Influence}: Sway points
				\item \textbf{Heteroskedasticity}: Uneven errors signal gaps
			\end{itemize}
			\begin{lstlisting}
# R (p. 177)
house_98105 <- house[house$ZipCode == 98105, ]
lm_98105 <- lm(AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms +
Bedrooms + BldgGrade, data = house_98105)
sresid <- rstandard(lm_98105)  # -4.326732 outlier
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{figure}
				\includegraphics[width=\textwidth]{figure4-6-placeholder.pdf}
				\caption{Influence Plot}
			\end{figure}
		\end{columns}
	\end{frame}
	
	\begin{frame}[fragile]{Diagnostics: Python}
		\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Identifies \$119,748 as partial sale anomaly
				\item \textbf{Key Insight}: Diagnostics ensure robust housing predictions
			\end{itemize}
			\begin{lstlisting}
# Python (p. 178)
from statsmodels.stats.outliers_influence import OLSInfluence
house_98105 = house[house['ZipCode'] == 98105]
model = smf.ols('AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms + Bedrooms + BldgGrade', data=house_98105).fit()
influence = OLSInfluence(model)
sresiduals = influence.resid_studentized
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Spots critical flaws
			\end{itemize}
		\end{columns}
	\end{frame}
	
	% Section 5: Insights & Next Steps
	\section{Insights \& Next Steps}
	
	\begin{frame}{Housing Insights}
		\begin{itemize}
			\item \textbf{Findings}: Linear ties price to size, splines capture nonlinear trends
			\item \textbf{Diagnostics}: Reveal quirks like partial sales—critical for accuracy
			\item \textbf{Application}: Real-world tool for buyers, sellers, and assessors
		\end{itemize}
	\end{frame}
	
	\begin{frame}{Key Takeaways}
		\begin{itemize}
			\item \textbf{Flexible}: Evolves from simple lines to complex curves for housing
			\item \textbf{Precise}: RMSE and cross-validation ensure reliable price predictions
			\item \textbf{Powerful}: R and Python implementations unlock data-driven insights
		\end{itemize}
	\end{frame}
	
	\begin{frame}{Looking Ahead}
		\begin{itemize}
			\item \textbf{Resources}: \textit{Statistical Learning} (Hastie et al.), \textit{Time Series Forecasting} (Shmueli)
			\item \textbf{Next Steps}: Dive into splines, time series for dynamic housing models
			\item \textbf{Call}: Blend theory and practice for smarter predictions
		\end{itemize}
	\end{frame}
	
\end{document}