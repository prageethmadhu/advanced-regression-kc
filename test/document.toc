\beamer@sectionintoc {1}{Starting Simple}{4}{0}{1}
\beamer@sectionintoc {2}{Expanding the Scope}{8}{0}{2}
\beamer@sectionintoc {3}{Refining Precision}{17}{0}{3}
\beamer@sectionintoc {4}{Facing Challenges}{24}{0}{4}
\beamer@sectionintoc {5}{Thank You... ! }{37}{0}{5}
