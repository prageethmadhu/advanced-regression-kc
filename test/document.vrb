\frametitle{Diagnostics: Python}
\lstset{language=Python}
		\begin{columns}
			\column{0.6\textwidth}
			\begin{itemize}
				\item \textbf{Practice}: Identifies \$119,748 as partial sale anomaly
				\item \textbf{Key Insight}: Diagnostics ensure robust housing predictions
			\end{itemize}
			\begin{lstlisting}
				# Python (p. 178)
				from statsmodels.stats.outliers_influence import OLSInfluence
				house_98105 = house[house['ZipCode'] == 98105]
				model = smf.ols('AdjSalePrice ~ SqFtTotLiving + SqFtLot + Bathrooms + Bedrooms + BldgGrade', data=house_98105).fit()
				influence = OLSInfluence(model)
				sresiduals = influence.resid_studentized
			\end{lstlisting}
			\column{0.4\textwidth}
			\begin{itemize}
				\item Spots critical flaws
			\end{itemize}
		\end{columns}
