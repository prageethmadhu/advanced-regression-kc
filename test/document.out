\BOOKMARK [2][]{Outline0.1}{\376\377\000S\000t\000a\000r\000t\000i\000n\000g\000\040\000S\000i\000m\000p\000l\000e}{}% 1
\BOOKMARK [2][]{Outline0.2}{\376\377\000E\000x\000p\000a\000n\000d\000i\000n\000g\000\040\000t\000h\000e\000\040\000S\000c\000o\000p\000e}{}% 2
\BOOKMARK [2][]{Outline0.3}{\376\377\000R\000e\000f\000i\000n\000i\000n\000g\000\040\000P\000r\000e\000c\000i\000s\000i\000o\000n}{}% 3
\BOOKMARK [2][]{Outline0.4}{\376\377\000F\000a\000c\000i\000n\000g\000\040\000C\000h\000a\000l\000l\000e\000n\000g\000e\000s}{}% 4
\BOOKMARK [2][]{Outline0.5}{\376\377\000T\000h\000a\000n\000k\000\040\000Y\000o\000u\000.\000.\000.\000\040\000!\000\040}{}% 5
